api/ocm/extensions/repositories/genericocireg/componentversion.go

```diff
@@ -183,32 +187,36 @@ func (c *ComponentVersionContainer) Update() (bool, error) {
 			layers.Add(i)
 		}
 		for i, r := range desc.Resources {
-			s, l, err := c.evalLayer(r.Access)
+			s, list, err := c.evalLayer(r.Access)
 			if err != nil {
 				return false, fmt.Errorf("failed resource layer evaluation: %w", err)
 			}
-			if l > 0 {
-				layerAnnotations[l] = append(layerAnnotations[l], ArtifactInfo{
-					Kind:     ARTKIND_RESOURCE,
-					Identity: r.GetIdentity(desc.Resources),
-				})
-				layers.Delete(l)
+			if len(list) > 0 {
```

Comment:
You don't need the length check if you are using a for loop.

Reply:
done